client/state/document/reducer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import filter from 'lodash/filter';
+
+/**
+ * Internal dependencies
+ */
+import {
+	DOCUMENT_LINK_ADD,
+	DOCUMENT_LINK_REMOVE,
+	DOCUMENT_META_ADD,
+	DOCUMENT_META_REMOVE,
+	DOCUMENT_DESCRIPTION_SET,
+	DOCUMENT_TITLE_SET,
+	DOCUMENT_UNREAD_COUNT_SET
+} from 'state/action-types';
+
+export function title( state = '', action ) {
+	switch ( action.type ) {
+		case DOCUMENT_TITLE_SET:
+			return action.title;
+	}
+
+	return state;
+}
+
+export function unread_count( state = 0, action ) {
+	switch ( action.type ) {
+		case DOCUMENT_UNREAD_COUNT_SET:
+			return action.count;
+	}
+
+	return state;
+}
+
+export function description( state = '', action ) {
+	switch ( action.type ) {
+		case DOCUMENT_DESCRIPTION_SET:
+			return action.description;
+	}
+
+	return state;
+}
+
+export function meta( state = [], action ) {
+	switch ( action.type ) {
+		case DOCUMENT_META_ADD:
+			return [...state, action.meta];
```

Comment:
ESLint doesn't capture it, but there should be spaces between square brackets of an array ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#arrays-and-function-calls)).